/fastlivo2_node:
  ros__parameters:
    common:
      img_topic: "/left_camera/image"
      lid_topic: "/livox/lidar"
      imu_topic: "/livox/imu"
      img_en: 1
      lidar_en: 1
      ros_driver_bug_fix: false

    extrin_calib:
      extrinsic_T: [0.04165, 0.02326, -0.0284]
      extrinsic_R: [1., 0., 0., 0., 1., 0., 0., 0., 1.]
      Rcl: [0.00610193,-0.999863,-0.0154172,
            -0.00615449,0.0153796,-0.999863,
            0.999962,0.00619598,-0.0060598] 
      Pcl: [0.0194384, 0.104689,-0.0251952]

    time_offset: 
      imu_time_offset: 0.0
      img_time_offset: 0.1
      exposure_time_init: 0.0

    preprocess:
      point_filter_num: 1
      filter_size_surf: 0.1
      lidar_type: 1 # Livox Avia LiDAR
      scan_line: 6
      blind: 0.8

    vio:
      max_iterations: 5
      outlier_threshold: 1000.0 # 78 100 156 #100 200 500 700 infinite
      img_point_cov: 100.0 # 100 1000 
      grid_size: 5
      grid_n_height: 5
      patch_size: 8
      patch_pyrimid_level: 4
      normal_en: true
      raycast_en: false
      inverse_composition_en: false
      exposure_estimate_en: true
      inv_expo_cov: 0.1

    imu:
      imu_en: true
      gravity_est_en: true
      imu_int_frame: 30
      acc_cov: 0.5 # 0.2
      gyr_cov: 0.3 # 0.5
      b_acc_cov: 0.0001 # 0.1
      b_gyr_cov: 0.0001 # 0.1

    lio:
      max_iterations: 5
      dept_err: 0.02
      beam_err: 0.05
      min_eigen_value: 0.0025 # 0.005
      voxel_size: 0.5
      max_layer: 2
      max_points_num: 50
      layer_init_num: [5, 5, 5, 5, 5]

    local_map:
      map_sliding_en: false
      half_map_size: 100
      sliding_thresh: 8

    uav:
      imu_rate_odom: false
      gravity_align_en: false

    publish:
      dense_map_en: true
      pub_effect_point_en: false 
      pub_plane_en: false
      pub_scan_num: 1
      blind_rgb_points: 0.0

    evo:
      seq_name: "CBD_Building_01"
      pose_output_en: false

    pcd_save:
      pcd_save_en: false
      colmap_output_en: false # need to set interval = -1
      filter_size_pcd: 0.15
      interval: -1
      # how many LiDAR frames saved in each pcd file;
      # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.

    camera:
      cam_model: Pinhole
      cam_width: 1280
      cam_height: 1024
      scale: 0.5
      cam_fx: 1293.56944
      cam_fy: 1293.3155
      cam_cx: 626.91359
      cam_cy: 522.799224
      cam_d0: -0.076160
      cam_d1: 0.123001
      cam_d2: -0.00113
      cam_d3: 0.000251
